api/jobs_test.go

```diff
@@ -681,3 +698,481 @@ func TestPutNumTasksHandler(t *testing.T) {
+
+			Convey("Then updatedJob and bsonUpdates should contain updates from the patch", func() {
+				So(updatedJob.State, ShouldEqual, models.JobStateCompleted)
+				So(bsonUpdates[models.JobStateBSONKey], ShouldEqual, models.JobStateCompleted)
+
+				Convey("And ReindexCompleted should be updated", func() {
+					So(updatedJob.ReindexCompleted, ShouldNotEqual, currentJob.ReindexCompleted)
+					So(bsonUpdates[models.JobReindexCompletedBSONKey], ShouldNotBeEmpty)
+
+					Convey("And LastUpdated should be updated", func() {
+						So(updatedJob.LastUpdated, ShouldNotEqual, currentJob.LastUpdated)
+						So(bsonUpdates[models.JobLastUpdatedBSONKey], ShouldNotBeEmpty)
+
+						Convey("And no error should be returned", func() {
+							So(err, ShouldBeNil)
+						})
+					})
+				})
+			})
+		})
+	})
+}
+
+func TestPreparePatchUpdatesFail(t *testing.T) {
+	t.Parallel()
+	testCtx := context.Background()
+
+	currentJob, err := models.NewJob(validJobID1)
+	if err != nil {
+		t.Error("failed to get job to test preparePatchUpdates")
+	}
+
+	Convey("Given patches with unknown patch operation", t, func() {
+		unknownOpPatches := []dprequest.Patch{
+			{
+				Op:   dprequest.OpRemove.String(),
+				Path: models.JobTotalSearchDocumentsPath,
+			},
+		}
+
+		Convey("When preparePatchUpdates is called", func() {
+			updatedJob, bsonUpdates, err := api.GetUpdatesFromJobPatches(testCtx, unknownOpPatches, currentJob, log.Data{})
+
+			Convey("And an error should be returned", func() {
+				So(err, ShouldNotBeNil)
+				So(err.Error(), ShouldEqual, fmt.Sprintf("patch operation '%s' not allowed, expected '%v'", unknownOpPatches[0].Op, models.ValidPatchOps))
+
+				So(updatedJob, ShouldResemble, models.Job{})
+				So(bsonUpdates, ShouldBeEmpty)
+			})
+		})
+	})
+
+	Convey("Given patches with unknown path", t, func() {
+		unknownPathPatches := []dprequest.Patch{
+			{
+				Op:    dprequest.OpReplace.String(),
+				Path:  "/unknown",
+				Value: "unknown",
+			},
+		}
+
+		Convey("When preparePatchUpdates is called", func() {
+			updatedJob, bsonUpdates, err := api.GetUpdatesFromJobPatches(testCtx, unknownPathPatches, currentJob, log.Data{})
+
+			Convey("And an error should be returned", func() {
+				So(err, ShouldNotBeNil)
+				So(err.Error(), ShouldEqual, fmt.Sprintf("provided path '%s' not supported", unknownPathPatches[0].Path))
+
+				So(updatedJob, ShouldResemble, models.Job{})
+				So(bsonUpdates, ShouldBeEmpty)
+			})
+		})
+	})
+
+	Convey("Given patches with invalid number of tasks", t, func() {
+		invalidNoOfTasksPatches := []dprequest.Patch{
+			{
+				Op:    dprequest.OpReplace.String(),
+				Path:  models.JobNoOfTasksPath,
+				Value: "unknown",
+			},
+		}
+
+		Convey("When preparePatchUpdates is called", func() {
+			updatedJob, bsonUpdates, err := api.GetUpdatesFromJobPatches(testCtx, invalidNoOfTasksPatches, currentJob, log.Data{})
+
+			Convey("And an error should be returned", func() {
```

Comment:
```suggestion
			Convey("Then an error should be returned", func() {
```
Please also change the ones below this to be 'Then'.